core/vm/fvm/fvm.cpp

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "vm/fvm/fvm.hpp"
+
+#include <filecoin-ffi/filcrypto.h>
+#include <shared_mutex>
+
+#include "cbor_blake/ipld_any.hpp"
+#include "cbor_blake/ipld_version.hpp"
+#include "codec/cbor/light_reader/cid.hpp"
+#include "common/ffi.hpp"
+#include "common/logger.hpp"
+#include "vm/fvm/ubig128.hpp"
+#include "vm/runtime/circulating.hpp"
+#include "vm/runtime/consensus_fault.hpp"
+#include "vm/runtime/runtime_randomness.hpp"
+#include "vm/state/impl/state_tree_impl.hpp"
+
+#define FFI_TRY(name)                                                          \
+  do {                                                                         \
+    if (res->status_code != FCPResponseStatus::FCPResponseStatus_FCPNoError) { \
+      spdlog::error(                                                           \
+          "{} status={} message={}", #name, res->status_code, res->error_msg); \
+      return ERROR_TEXT(#name);                                                \
+    }                                                                          \
+  } while (0)
```

Comment:
Maybe `while (true) ` is better?

Reply:
No, https://gcc.gnu.org/onlinedocs/cpp/Swallowing-the-Semicolon.html